pandas/core/dtypes/concat.py

```diff
@@ -1,31 +1,82 @@
 """
 Utility functions related to concat.
 """
-from typing import cast
+from typing import Any, List, cast
 
 import numpy as np
 
 from pandas._typing import ArrayLike, DtypeObj
 
-from pandas.core.dtypes.cast import find_common_type
+from pandas.core.dtypes.cast import ensure_dtype_can_hold_na, find_common_type
 from pandas.core.dtypes.common import (
     is_categorical_dtype,
     is_dtype_equal,
     is_extension_array_dtype,
     is_sparse,
 )
 from pandas.core.dtypes.generic import ABCCategoricalIndex, ABCSeries
+from pandas.core.dtypes.missing import na_value_for_dtype
 
 from pandas.core.arrays import ExtensionArray
 from pandas.core.arrays.sparse import SparseArray
 from pandas.core.construction import array, ensure_wrapped_if_datetimelike
 
 
+class NullArrayProxy:
```

Comment:
another option if we dont want a whole new thing would be to use `np.empty(shape, dtype="V")`  (which has obj.nbytes = 0)